### DISCUSSION TITLE
How to get the name of the toolchain `size` command?

### DISCUSSION THREAD

**Discussion by ilg-ul** [mesonbuild/meson#9913] [Q&A] [answered]
I have toolchain definitions like:

```
arm-none-eabi-gcc.ini

[binaries]
c = 'arm-none-eabi-gcc'
cpp = 'arm-none-eabi-g++'
strip = 'arm-none-eabi-strip'
ar = 'arm-none-eabi-gcc-ar'
ranlib = 'arm-none-eabi-gcc-ranlib'
size = 'arm-none-eabi-size'
```

and similarly for Windows:

```
arm-none-eabi-gcc-cmd.ini

[binaries]
c = 'arm-none-eabi-gcc.cmd'
cpp = 'arm-none-eabi-g++.cmd'
strip = 'arm-none-eabi-strip.cmd'
ar = 'arm-none-eabi-gcc-ar.cmd'
ranlib = 'arm-none-eabi-gcc-ranlib.cmd'
size = 'arm-none-eabi-size.cmd'
```

In order to run the `size` command within a custom target, I need to get the actual command name, regardless the build platform.

I checked the functions available in the `compiler` and `meson` objects, but I could not identify one to return the command associated with one tool.

Any suggestions?

**Comment by eli-schwartz** [comment, marked as answer]
The binaries table overrides the result of `find_program('size')`, and *additionally* will be used when Meson internally looks up programs as part of the compiler toolchain handling.

https://mesonbuild.com/Machine-files.html#binaries
> The binaries section contains a list of binaries. These can be used internally by Meson, or by the `find_program` function.

**Reply by ilg-ul** [reply]
I confirm, `find_program('size')` returns the name of the command as defined in the `cross.ini` file.

Thank you!
